### PR TITLE
Task 46/49: BE: Implement POST API to add product to the shopping cart/GET API to get shopping cart products list

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/cart/cart.controller.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  constructor(private readonly cartService: CartService) {}
+
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth('JWT-auth')
+  @Get('cart')
+  @ApiOperation({ summary: SummaryConstants.GET_CART})
+  async getCart(@Request() req) {
+    const userId = req.user._id;
+
+    return this.cartService.getCart(userId);
```
Comment (VYaniv) [inline src/cart/cart.controller.ts]: `return this.cartService.getCart(req.user._id)`

File: src/cart/cart.controller.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth('JWT-auth')
+  @Post('add')
+  async addToCart(@Request() req, @Body() addToCartDto: AddToCartDto) {
+    const userId = req.user._id;
+
+    if (!userId) {
+      throw new HttpException('User ID is required', HttpStatus.BAD_REQUEST);
+    }
```
Comment (VYaniv) [inline src/cart/cart.controller.ts]: When is it possible?

**Review by VYaniv — APPROVED** [review]

File: src/cart/cart.controller.ts (hunk 1 above)
Comment (di-kinosian) [reply src/cart/cart.controller.ts]: done

File: src/cart/cart.controller.ts (hunk 2 above)
Comment (di-kinosian) [reply src/cart/cart.controller.ts]: done

**Pull request merged**